pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -257,3 +266,65 @@ func (c *cephStatusChecker) conditionMessageReason(condition cephv1.ConditionTyp
 
 	return condition, reason, message
 }
+
+// forceDeleteStuckPodsOnNotReadyNodes lists all the nodes that are in NotReady state and
+// gets all the pods on the failed node and force delete the pods stuck in terminating state.
+func (c *cephStatusChecker) forceDeleteStuckRookPodsOnNotReadyNodes() error {
+	nodes, err := k8sutil.GetNotReadyKubernetesNodes(c.context.Clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to get NotReady nodes")
+	}
+	for _, node := range nodes {
+		pods, err := c.getRookPodsOnNode(node.Name)
+		if err != nil {
+			logger.Errorf("failed to get pods on NotReady node %q. %v", node.Name, err)
+		}
+		for _, pod := range pods {
+			if err := k8sutil.ForceDeletePodIfStuck(c.context, pod); err != nil {
+				logger.Errorf("skipping forced delete of stuck pod %q. %v", pod.Name, err)
```

Comment:
This is the content of the preexisting `ForceDeletePodIfStuck`. It reports a warning if it isn't able to delete a stuck pod. It fails under only one condition: it can't get node info. @travisn should this function also return an error if it fails to delete the pod?

```go
func ForceDeletePodIfStuck(clusterdContext *clusterd.Context, pod v1.Pod) error {
	ctx := context.TODO()
	logger.Debugf("checking if pod %q is stuck and should be force deleted", pod.Name)
	if pod.DeletionTimestamp.IsZero() {
		logger.Debugf("skipping pod %q restart since the pod is not deleted", pod.Name)
		return nil
	}
	node, err := clusterdContext.Clientset.CoreV1().Nodes().Get(ctx, pod.Spec.NodeName, metav1.GetOptions{})
	if err != nil {
		return errors.Wrap(err, "node status is not available")
	}
	if NodeIsReady(*node) {
		logger.Debugf("skipping restart of pod %q since the node status is ready", pod.Name)
		return nil
	}

	logger.Infof("force deleting pod %q that appears to be stuck terminating", pod.Name)
	var gracePeriod int64
	deleteOpts := metav1.DeleteOptions{GracePeriodSeconds: &gracePeriod}
	if err := clusterdContext.Clientset.CoreV1().Pods(pod.Namespace).Delete(ctx, pod.Name, deleteOpts); err != nil {
		logger.Warningf("pod %q deletion failed. %v", pod.Name, err)
		return nil  // <---- MAYBE RETURN ERROR HERE???
	}
	logger.Infof("pod %q deletion succeeded", pod.Name)
	return nil
}
```

Other uses `ForceDeletePodIfStuck` report a warning after, not an error. We should probably be consistent.
```suggestion
				logger.Warningf("skipping restart of stuck pod %q. %v", pod.Name, err)
```

Reply:
For now, added the suggested change.